handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -305,6 +305,25 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         }
     }
 
+    /**
+     * Reset the read timeout. As this handler is not thread-safe, this method <b>must</b> be called on the event loop.
+     */
+    public void resetReadTimeout() {
+        if (readerIdleTimeNanos > 0 || allIdleTimeNanos > 0) {
+            lastReadTime = ticksInNanos();
+            reading = false;
+        }
+    }
+
+    /**
+     * Reset the write timeout. As this handler is not thread-safe, this method <b>must</b> be called on the event loop.
+     */
+    public void resetWriteTimeout() {
+        if (writerIdleTimeNanos > 0 || allIdleTimeNanos > 0) {
+            lastWriteTime = ticksInNanos();
```

Comment:
I think it this case you either need to offload the modifications to the `EventExecutor` or everything has to be volatile.

Reply:
i am assuming the method is called on the EventExecutor. unfortunately there is no EventExecutor available to verify / ensure this. I could store it in a field in handlerAdded, but I didn't want to add overhead for a feature most people won't use.

Reply:
Usually when handlers have such methods exposed these are thread-safe (like for example the methods on SslHandler).

Reply:
i agree that it would be better. however this would require adding an EventExecutor field and setting it in handlerAdded. It's minor additional overhead, but I felt the change would be better if the "default path" (that doesn't use the new methods) remained unchanged. If you disagree with that, I will change it.

Reply:
There is another issue I just realized with calling this method from outside the event loop. If you do something like this:

```java
handler.resetReadTimeout();
// T1
loop.execute(() -> { /* T2 */ });
```

It is actually possible that a read timeout happens *after* T1 but before T2. 

If the API forbids calls outside the event loop as it does now, this code now has to be:

```java
// T1
loop.execute(() -> { handler.resetReadTimeout(); /* T2 */ });
```

It is more obvious that a read timeout can happen after T1. While with your proposed change both code samples are functionally equivalent (both run resetReadTimeout on the event loop, first just does it implicitly), the second case makes it clear that the reset action is not immediate.